content/docs/building-apps/add-stellar-assets.mdx

```diff
@@ -0,0 +1,185 @@
+ *
+ * @param  {string} to      The account address of the recipient ("G...")
+ * @param  {Asset}  asset   The asset to credit from the source customer
+ * @param  {string} amount  The amount of `asset` to credit
+ *
+ * @return {xdr.paymentOp}
+ */
+function createPayment(to, asset, amount) {
+  return sdk.Operation.payment({
+    source: CUSTODIAL_KEY.publicKey(),
+    destination: to,
+    asset: asset,
+    amount: amount,
+  });
+}
+```
+
+</CodeExample>
+
+
+## Integration points for listing XLM
+
+### Setting the memo required flag on your account
+Before we get into the main integration points for adding XLM, make sure to set the memo required flag on your pooled account. This step is necessary to prevent customers from forgetting to attach the memo, which can result in additional support needed from the custodial services and a less desirable user experience for the new customers. This [article](https://www.stellar.org/developers-blog/fixing-memo-less-payments) explains how to set that up.
+
+### Listening for incoming payments from the Stellar network
+First, you need to listen for payments to the pooled account and credit any user that receives XLM there. For every payment received by the pooled account:
+
+  * check the memo field to determine which user should receive the payment, and
+  * debit the user’s account with the amount of XLM they received.
+
+This is the role of `depositIntoPool`, described above. You pass this function as the `onmessage` option when you [stream payments](https://developers.stellar.org/docs/tutorials/follow-received-payments/):
+
+<CodeExample>
+
+```js
+
+  const stream = server.payments()
+    .forAccount(custodialAcc)
+    .join("transactions")
+    .stream({
+      onmessage: (payment) => {
+        const from = payment.source_account;
+        const customerId = payment.memo;
+        const amount = payment.amount;
+
+        let asset = sdk.Asset.native();
+        if (payment.asset_issuer) {
+          asset = new sdk.Asset(payment.asset_code, payment.asset_issuer);
+        }
+
+        depositIntoPool(from, customerId, asset, amount);
+      }
+    });
+```
+
+</CodeExample>
+
+When someone **outside** of your customer base wants to send lumens to one of your customers, instruct them to make an XLM payment to your pooled account address with the customer ID in the memo field of the transaction. Assuming you have set the memo required flag on your account (see [above](#setting-the-memo-required-flag-on-your-account)), [well-behaved](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0029.md) wallets should enforce it and prevent users from forgetting to attach a memo to an incoming payment. However, to be on the safe side, you should make it incredibly clear to senders that their payment will end up in limbo if they fail to attach a valid one. You can learn more about the transaction memo field [here](https://developers.stellar.org/docs/glossary/transactions/#memo).
```

Comment:
Side note: It might be nice to have a sep (maybe there already is one), that enables a way to test if a "sub-account" exists or not (to check the memo field is valid), before sending a transfer to a pooled account.

Reply:
Yeah, I'm actually surprised that a SEP doesn't exist for managing pooled accounts in general. Maybe it's to avoid imposing requirements on custodians?